packages/venia-concept/src/RootComponents/Product/Product.js

```diff
@@ -1,96 +1,151 @@
+                        description: string,
+                        short_description: string,
+                        canonical_url: string
+                    })
+                ).isRequired
+            }).isRequired
         })
     };
 
-    static defaultProps = {
-        data: mockData
-    };
-
     render() {
-        const { classes, data } = this.props;
+        const { classes } = this.props;
 
         return (
             <Page>
-                <article className={classes.root}>
-                    <section className={classes.title}>
-                        <h1 className={classes.productName}>
-                            <span>{data.name}</span>
-                        </h1>
-                        <p className={classes.productPrice}>
-                            <span>{data.price}</span>
-                        </p>
-                    </section>
-                    <section className={classes.imageCarousel}>
-                        <Carousel images={data.images} />
-                    </section>
-                    <section className={classes.actions}>
-                        <button className={classes.action}>
-                            <span>Add to Wishlist</span>
-                        </button>
-                    </section>
-                    <section className={classes.options}>
-                        <Options options={data.options} />
-                    </section>
-                    <section className={classes.quantity}>
-                        <h2 className={classes.quantityTitle}>
-                            <span>Quantity</span>
-                        </h2>
-                        <Quantity />
-                    </section>
-                    <section className={classes.cartActions}>
-                        <button className={classes.addToCart}>
-                            <span>Add to Cart</span>
-                        </button>
-                    </section>
-                    <section className={classes.description}>
-                        <h2 className={classes.descriptionTitle}>
-                            <span>Product Description</span>
-                        </h2>
-                        <RichText content={data.description} />
-                    </section>
-                    <section className={classes.details}>
-                        <h2 className={classes.detailsTitle}>
-                            <span>Details</span>
-                        </h2>
-                        <RichText content={data.additionalInfo} />
-                    </section>
-                    <section className={classes.related}>
-                        <h2 className={classes.relatedTitle}>
-                            <span>Similar Products</span>
-                        </h2>
-                    </section>
-                </article>
+                <Query
+                    query={productDetailQuery}
+                    variables={{ urlKey: getUrlKey() }}
```

Comment:
Can we add a `TODO` that links to an issue on the [`graphql-ce`](https://github.com/magento/graphql-ce) repo with a request to add a filter by entity ID?

Reply:
I put it above the query definition.